src/main/java/org/spongepowered/common/mixin/plugin/tileentityactivation/TileEntityActivation.java

```diff
@@ -177,9 +176,13 @@ private static void activateChunkTileEntities(final EntityPlayer player, final C
                 }
 
                 final int bbActivationRange = ((ActivationCapability) tileEntity).activation$getActivationRange();
-                final int blockDistance = Math.round(tilePos.distance(playerPos));
-                if (blockDistance <= bbActivationRange) {
-                    ((ActivationCapability) tileEntity).activation$setActivatedTick(currentTick);
+                for (EntityPlayerMP player : players) {
+                    final Vector3i playerPos = VecHelper.toVector3i(player.getPosition());
+                    final int blockDistance = Math.round(tilePos.distance(playerPos));
```

Comment:
While we're looking at performance, I wonder if we should also look at making this dependent on distance squared. I realise that it makes some difference to getting `bbActivationRange` and `Math.round()` - but the activation range would then be `Math.pow(bbActivationRange - 0.5, 2)` due to the current rounding, and such rounding would not be required.

Reply:
Good point.
I've switched to `Vector3i#distanceSquared` (which returns an int, so there's no need for rounding).